packages/widget_toolkit_biometrics/lib/src/lib_biometrics/view/biometrics_switch.dart

```diff
@@ -0,0 +1,141 @@
+
+  /// [onError] is optional function that enable error handling out of the package
+  final void Function(ErrorModel)? onError;
+
+  @override
+  Widget build(BuildContext context) => RxBlocProvider<BiometricsBlocType>(
+        /*Todo(Toncho): this creates an implicit dependency
+           on [BiometricsService], */
+        create: (ctx) => BiometricsBloc(context.read<BiometricsService>()),
+        child: Column(
+          children: [
+            RxBlocBuilder<BiometricsBlocType, bool>(
+              state: (bloc) => bloc.states.areBiometricsEnabled,
+              builder: (context, enabled, bloc) {
+                final areEnabled = enabled.hasData && (enabled.data ?? false);
+                return builder?.call(
+                      areEnabled,
+                      (enable) => bloc.events
+                          .setBiometrics(enable, _getMessage(enable)),
+                    ) ??
+                    Switch(
+                      value: areEnabled,
+                      onChanged: (val) {
+                        bloc.events.setBiometrics(val, _getMessage(val));
+                      },
+                    );
+              },
+            ),
+            _showBottomMessage(context),
+            _buildErrorListener(context),
+          ],
+        ),
+      );
+
+  String _getMessage(bool isEnabled) =>
+      isEnabled ? (enabledMessage ?? activateBiometrics) : deactivateBiometrics;
+
+  Widget _buildErrorListener(BuildContext context) =>
+      RxBlocListener<BiometricsBlocType, ErrorModel>(
+        state: (bloc) => bloc.states.errors,
+        listener: (context, state) => onError?.call(state),
+      );
+
+  Widget _showBottomMessage(BuildContext context) =>
+      RxBlocListener<BiometricsBlocType, BiometricsSettingMessageType?>(
+        state: (bloc) => bloc.states.biometricsDialog,
+        listener: (context, state) => state != null
+            ? showDefaultNotification
+                ? showBiometricsMessageBottomSheet(context, state)
+                : onStateChanged?.call(state)
+            : null,
+      );
+
+  Future<void> showBiometricsMessageBottomSheet(
+    BuildContext context,
+    BiometricsSettingMessageType messageType,
+  ) =>
+      showBlurredBottomSheet(
+        context: context,
+        configuration: const ModalConfiguration(safeAreaBottom: false),
+        builder: (context) => Padding(
+          padding:
+              EdgeInsets.only(bottom: MediaQuery.of(context).viewInsets.bottom),
+          child: Container(
+            padding: const EdgeInsets.all(16),
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: [
+                MessagePanelWidget(
+                  message: mapMessageToString?.call(messageType) ??
+                      messageType.translate(),
+                  messageState: messageType.state(),
+                ),
+                const SizedBox(
```

Comment:
Please use the theme extension instead of hardcoding values in the widget layer

Reply:
The widget_toolkit library's [showBlurredBottomSheet] adds a properly themed and padded close button by default, I've removed the duplicate close button in the following commits